docs/upload-strategies/transloadit.mdx

```diff
@@ -0,0 +1,585 @@
+    // Shown after uploads have succeeded, but when the Assembly is still executing.
+    // This only shows if `waitForMetadata` or `waitForEncoding` was enabled.
+    encoding: 'Encoding...',
+  },
+}
+```
+
+<details>
+  <summary>Deprecated options</summary>
+
+These options have been deprecated in favor of [`assemblyOptions`](#assemblyOptions),
+which we now recommend for all use cases.
+You can still use these options, but they will be removed in the next major version.
+
+#### `params`
+
+The Assembly parameters to use for the upload (`Object`, default: `null`)
+See the Transloadit documentation on [Assembly Instructions](https://transloadit.com/docs/#14-assembly-instructions) for further information.
+
+The `auth.key` Assembly parameter is required. You can also use the `steps` or `template_id` options here as described in the Transloadit documentation.
+
+```js
+uppy.use(Transloadit, {
+  params: {
+    auth: { key: 'YOUR_TRANSLOADIT_KEY' },
+    steps: {
+      encode: {
+        robot: '/video/encode',
+        use: {
+          steps: [':original'],
+          fields: ['file_input_field2'],
+        },
+        preset: 'iphone',
+      },
+    },
+  },
+})
+```
+
+#### `signature`
+
+An optional signature for the Assembly parameters.
+See the Transloadit documentation on [Signature Authentication](https://transloadit.com/docs/#26-signature-authentication) for further information.
+
+If a `signature` is provided, `params` should be a JSON string instead of a JavaScript object,
+as otherwise the generated JSON in the browser may be different from the JSON string that was used to generate the signature.
+
+#### `fields`
+
+An object of form fields to send along to the Assembly.
+Keys are field names, and values are field values.
+See also the Transloadit documentation on [Form Fields In Instructions](https://transloadit.com/docs/#23-form-fields-in-instructions).
+
+```js
+uppy.use(Transloadit, {
+  // ...
+  fields: {
+    message: 'This is a form field',
+  },
+})
+```
+
+You can also pass an array of field names to send global or file metadata along to the Assembly.
+Global metadata is set using the [`meta` option](/docs/uppy/#meta) in the Uppy constructor,
+or using the [`setMeta` method](/docs/uppy/#uppy-setMeta-data).
+File metadata is set using the [`setFileMeta`](/docs/uppy/#uppy-setFileMeta-fileID-data) method.
+The [Form](/docs/form) plugin also sets global metadata based on the values of `<input />`s in the form,
+providing a handy way to use values from HTML form fields:
+
+```js
+uppy.use(Form, { target: 'form#upload-form', getMetaFromForm: true })
+uppy.use(Transloadit, {
+  fields: ['field_name', 'other_field_name'],
+  params: { /* ... */ },
+})
+```
+
+Form fields can also be computed dynamically using custom logic, by using the [`getAssemblyOptions(file)`](/docs/transloadit/#getAssemblyOptions-file) option.
+
+</details>
+
+### Static properties
+
+#### `Transloadit.COMPANION`
```

Comment:
This is deprecated, users should use `import {COMPANION_URL} from '@uppy/transloadit'`